server/handlers/user.go

```diff
@@ -103,3 +103,34 @@ func (s *Server) loginImplicitly(w http.ResponseWriter, res *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(user)
 }
+
+func (s *Server) editProfile(w http.ResponseWriter, req *http.Request) {
+
+	verifiedToken := req.Context().Value(ContextKey).(*jwt.Token)
+
+	subject, tokenErr := verifiedToken.Claims.GetSubject()
+	if tokenErr != nil {
+		//not sure what error msg is best here
+		respondWithError(w, "Invalid Token", http.StatusBadRequest)
+		return
+	}
+
+	user := data.UserProfile{}
+	decodeErr := json.NewDecoder(req.Body).Decode(&user)
+	if decodeErr != nil {
+		respondWithError(w, "Internal Server Error", http.StatusInternalServerError)
```

Comment:
If the request's body didn't match the struct we expected it's not the server's fault, it's the client's fault. Maybe we can change this status code to bad request?